metrics/web/test/dashboard/presentation/widgets/build_number_scorecard_test.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:metrics/base/presentation/widgets/scorecard.dart';
+import 'package:metrics/common/presentation/metrics_theme/model/metric_widget_theme_data.dart';
+import 'package:metrics/common/presentation/metrics_theme/model/metrics_theme_data.dart';
+import 'package:metrics/dashboard/presentation/view_models/build_number_scorecard_view_model.dart';
+import 'package:metrics/dashboard/presentation/widgets/build_number_scorecard.dart';
+import 'package:metrics/dashboard/presentation/widgets/no_data_placeholder.dart';
+
+import '../../../test_utils/metrics_themed_testbed.dart';
+
+void main() {
+  group("BuildNumberTextMetric", () {
+    testWidgets(
+      'displays the `no data placeholder` if the given build number metric is null',
+      (tester) async {
+        await tester.pumpWidget(const _BuildNumberScorecardTestbed(
+          buildNumberViewModel: null,
+        ));
+
+        expect(find.byType(NoDataPlaceholder), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      'displays the `no data placeholder` if the given number of builds is equal to 0',
+      (tester) async {
+        await tester.pumpWidget(const _BuildNumberScorecardTestbed(
+          buildNumberViewModel:
+              BuildNumberScorecardViewModel(numberOfBuilds: 0),
+        ));
+
+        expect(find.byType(NoDataPlaceholder), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "displays the Scorecard widget with the given number of builds as a value",
```

Comment:
Should we also add a test for description? To make sure it's DashboardStrings.perWeek?